### ISSUE TITLE
Default section in sidebar not working

### ISSUE THREAD

**Issue by lexislav** [barrelstrength/craft-sprout-forms#446]
### Description

The form section that will be selected by default in the sidebar navigation is not working. It always jumps to Forms, although Entries section is selected in settings.

_EDIT: this also leads to error, when user has sprout forms permissions for entries only: plugin jumps to forms section and tells user is is not authorized :-/_

### Additional info

- Plugins & versions: PRO 3.8.5
- PHP version	7.2.28
- OS version	Linux 4.15.0-65-generic
- Database driver & version	MySQL 5.5.5
- Image driver & version	Imagick 3.4.4 (ImageMagick 6.9.7-4)
- Craft edition & version	Craft Pro 3.4.9

**Comment by BenParizek** [comment]
@lexislav In my tests the behavior is working as expected for simple use cases so I'm currently thinking that you are seeing odd behavior because of some combination of settings you have in place that we may not be planning for correctly. Below is the logic that's happening in Sprout Forms when choosing which section to redirect to, with some added comments. Can you confirm what settings you have in place as it relates to Form and Entry permissions and Saving Data? Do you see anywhere in this logic where you may be getting redirected to the Forms section when you expect to be redirected to the Entries section?

``` php
// Does the user have permission to View Entries? Does Sprout Forms have permission to save data to the database? If either is false, we will not show the Entries tab.
$canViewEntries = Craft::$app->getUser()->checkPermission('sproutForms-viewEntries') &&
            $settings->enableSaveData;

// If the user has permission to access the Entries Section, and the Default Section setting is set to Entries, redirect to the entries section
if ($canViewEntries && $settings->defaultSection === 'entries') {
    return Craft::$app->getResponse()->redirect(UrlHelper::cpUrl('sprout-forms/entries'));
}

// If not, redirect to the Forms section
return Craft::$app->getResponse()->redirect(UrlHelper::cpUrl('sprout-forms/forms'));
```

**Comment by lexislav** [comment]
That is odd. The logic You are have send seems fine to me.
Well, but it dosn work even for administrator. In my case. 
Sprout Forms has permissions to save to DB, not the spam.
The Default section (Výchozí sekce) is translated… to Formuláře (Forms) and Záznamy (Entries). (Might it be related to translations?)



The entries section is divided to Web groups… 

Sprout SEO is installed as well.

If it helps, I might send You a project config or whatever.

Thansks

Alexandr

**Comment by BenParizek** [comment]
@lexislav I don't think the project config will help in this case cause it seems to be something in the database that's not creating the same logic.

I might try to delete and rebuild the `vendor/` folder to make sure all the latest dependencies are in place.

The other question I have is that if your logic is always defaulting to the forms tab, it seems that the `$canViewEntries && $settings->defaultSection === 'entries'` is always returning false. 

If you are comfortable doing so, you could add a `\Craft::dd()` statement to see what value is being returned for each of those variables in logic in the FormsController:
https://github.com/barrelstrength/craft-sprout-forms/blob/v3.8.7/src/controllers/FormsController.php#L57

```
\Craft::dd(Craft::$app->getUser()->checkPermission('sproutForms-viewEntries'));
\Craft::dd($settings->enableSaveData);
\Craft::dd($settings->defaultSection === 'entries');
```

It may also be worth asking what action you are taking to see the redirect happen incorrectly (to make sure we're testing the same thing). In my tests, I am clicking on the top-level "Sprout Forms" link in the sidebar which routes to `/admin/sprout-forms` and then processes that logic to redirect to one of the two pages (entries or forms).

**Comment by lexislav** [comment]
Hi
at first - yes we are talking about the same action. I am clicking on the top-level “Sprout Forms“ on the sidebar, which opens the forms and Sould open the Entries (in my setting).

I have entered those three lines to the controller. Not sure if correctly - my php skills are limite to echo something :-/

Then when I click on “Sprout forms“ in CP, I got white screen with single “True“ on it.
Does it help?

**Comment by BenParizek** [comment]
@lexislav Hah. Well, those three lines are three separate lines that you'd probably have to add one by one. Each one kills the PHP script and tells you the value of the variable that it defines.

So, where Sprout defines the line:

```
$canViewEntries = Craft::$app->getUser()->checkPermission('sproutForms-viewEntries') &&
            $settings->enableSaveData;
```

You can kill the script after that to see what each of those values are:

```
$canViewEntries = Craft::$app->getUser()->checkPermission('sproutForms-viewEntries') &&
            $settings->enableSaveData;

\Craft::dd(Craft::$app->getUser()->checkPermission('sproutForms-viewEntries'));
```

AND

```
$canViewEntries = Craft::$app->getUser()->checkPermission('sproutForms-viewEntries') &&
            $settings->enableSaveData;

\Craft::dd($settings->enableSaveData);
```
If you refresh after adding each of those lines separately, the true and false values you see will help us understand what value Sprout thinks it's processing in that logic.

In addition, we can check to see what value Sprout thinks the redirect setting is. This line should also return true or false to confirm you have the 'entries' option selected in your settings (or at least confirm what Sprout thinks that setting is):

```
\Craft::dd($settings->defaultSection === 'entries');
```

**Comment by lexislav** [comment]
Got it.

\Craft::dd(Craft::$app->getUser()->checkPermission('sproutForms-viewEntries'));
True

\Craft::dd($settings->enableSaveData);
‚1‘


\Craft::dd($settings->defaultSection === 'entries');
false

\Craft::dd($settings->defaultSection);
‚záznamy‘

Here we go.
The value of default Section got translated.
So, I have tried to reset the translations file sprout-forms.php, line 33, ‚Entries‘ => ‚Záznamy‘ to ‚Entries‘ => ‚Entries‘ and Bingo! It works.
But I still woul like to have Entries translated, because of editors…

Thanks for the patience.

Alexandr

**Comment by BenParizek** [comment]
@lexislav Okay, I see what's going on now. Thanks for helping with those debugging statements. 

I don't think your "Entries" label translation is the issue. I think the issue relates to our code accidentally wrapping the `entries` and `forms` slugs in translation tags. You should be able to translate the label, not the slug.

I've just pushed what I think will fix the issue to the `develop` branch. You can see the commit here:
https://github.com/barrelstrength/craft-sprout-forms/commit/2856a9b579468dcdaeef8b68696b47f1c6a3360c

I will push that update with the next release but if you'd like to manually apply those changes to the files you see listed in the commits, I believe that will solve the issue you are seeing and you can add back your translation.

**Comment by lexislav** [comment]
Yes, it works. Thanks!
To resave the settings was neccessary, but it si working now.

**Comment by BenParizek** [comment]
This fix is now released in Sprout Forms v3.9.0
